task.sql

```diff
@@ -9,25 +7,48 @@ CREATE TABLE Countries (
     PRIMARY KEY (ID)
 );
 
-CREATE TABLE ProductInventory (
-    ID INT,
-    ProductName VARCHAR(50),
-    WarehouseAmount INT,
+CREATE TABLE Warehouses (
+    WarehouseID INT,
     WarehouseName VARCHAR(50),
-    WarehouseAddress VARCHAR(50), 
+    WarehouseAddress VARCHAR(50),
     CountryID INT,
-	FOREIGN KEY (CountryID) REFERENCES Countries(ID) ON DELETE NO ACTION,
-    PRIMARY KEY (ID)
+    PRIMARY KEY (WarehouseID),
+    FOREIGN KEY (CountryID) REFERENCES Countries(ID) ON DELETE NO ACTION
+);
+
+CREATE TABLE Products (
+    ProductID INT,
+    ProductName VARCHAR(50),
+    PRIMARY KEY (ProductID)
+);
+
+CREATE TABLE ProductInventory (
+    WarehouseAmount INT,
+    WarehouseID INT,
+    ProductID INT,
+    PRIMARY KEY (ProductID, WarehouseID),
+    FOREIGN KEY (WarehouseID) REFERENCES Warehouses(WarehouseID) ON DELETE NO ACTION,
+    FOREIGN KEY (ProductID) REFERENCES Products(ProductID) ON DELETE NO ACTION
 );
 
--- Populate test data
-
-INSERT INTO Countries (ID,Name)
-	VALUES (1, 'Country1');
-INSERT INTO Countries (ID,Name)
-	VALUES (2, 'Country2');
-    
-INSERT INTO ProductInventory (ID,ProductName,WarehouseAmount,WarehouseName,WarehouseAddress,CountryID)
-	VALUES (1, 'AwersomeProduct', 2, 'Warehouse-1', 'City-1, Street-1',1);
-INSERT INTO ProductInventory (ID,ProductName,WarehouseAmount,WarehouseName,WarehouseAddress,CountryID)
-	VALUES (2, 'AwersomeProduct', 5, 'Warehouse-2', 'City-2, Street-2',2);
+INSERT INTO Countries (ID, Name)
+VALUES (1, 'Country1'), (2, 'Country2');
+
+INSERT INTO Warehouses (WarehouseID, WarehouseName, WarehouseAddress, CountryID)
+VALUES 
+    (1, 'Warehouse-1', 'City-1, Street-1', 1),
+    (2, 'Warehouse-2', 'City-2, Street-2', 2);
+
+INSERT INTO Products (ProductID, ProductName)
+VALUES 
+    (1, 'AwersomeProduct'),
+    (2, 'AwersomeProduct'),
+    (3, 'AwersomeProduct'),
+    (4, 'AwersomeProduct');
```

Comment:
The `Products` table has duplicate product names ('AwersomeProduct') with different ProductIDs. Ensure that this is intentional and aligns with your business logic. If product names should be unique, consider adding a UNIQUE constraint to the `ProductName` column.